File: src/VscPreviewExtension/Microsoft.DocAsCode.Dfm.VscPreview/TokenTreeProcessor.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.DocAsCode.Dfm.VscPreview
{
    using Microsoft.DocAsCode.Build.Engine;
    using Microsoft.DocAsCode.Plugins;

    public class TokenTreeProcessor
    {
        private static readonly DfmJsonTokenTreeServiceProvider DfmJsonTokenTreeServiceProvider = new DfmJsonTokenTreeServiceProvider();
        private static readonly IMarkdownService DfmMarkdownService = DfmJsonTokenTreeServiceProvider.CreateMarkdownService(new MarkdownServiceParameters());

        public static string TokenTreePreview(string rawMarkdownContent)
        {
            return DfmMarkdownService.Markup(rawMarkdownContent, null).Html;
        }
    }
}

File: tools/VscPreviewExtension/src/Processors/tokenTreeProcessor.ts
// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { ExtensionContext, window } from "vscode";

import { PreviewProcessor } from "./previewProcessor";
import { ProxyResponse } from "../Proxy/proxyResponse";
import { TokenTreeContentProvider } from "../ContentProvider/tokenTreeContentProvider";

export class TokenTreeProcessor extends PreviewProcessor {
    provider: TokenTreeContentProvider;

    constructor(context: ExtensionContext) {
        super(context);
        this.provider = new TokenTreeContentProvider(context);
    }

    protected pageRefresh(response: ProxyResponse){
        this.provider.update(response.documentUri, response.markupResult);
    }
}
